scipy/stats/_distribution_infrastructure.py

```diff
@@ -0,0 +1,5082 @@
+        >>> plt.show()
+
+        Plot the PDF of the normal distribution as a function of the
+        CDF for various values of the scale parameter.
+
+        >>> X = stats.Normal(mu=0., sigma=[0.5, 1., 2])
+        >>> X.plot('cdf', 'pdf')
+        >>> plt.show()
+
+        """
+
+        # Strategy: given t limits, get quantile limits. Form grid of
+        # quantiles, compute requested x and y at quantiles, and plot.
+        # Currently, the grid of quantiles is always linearly spaced.
+        # Instead of always computing linearly-spaced quantiles, it
+        # would be better to choose:
+        # a) quantiles or probabilities
+        # b) linearly or logarithmically spaced
+        # based on the specified `t`.
+        # TODO:
+        # - smart spacing of points
+        # - when the parameters of the distribution are an array,
+        #   use the full range of abscissae for all curves
+
+        t_is_quantile = {'x', 'icdf', 'iccdf', 'ilogcdf', 'ilogccdf'}
+        t_is_probability = {'cdf', 'ccdf', 'logcdf', 'logccdf'}
+        valid_t = t_is_quantile.union(t_is_probability)
+        valid_xy =  valid_t.union({'pdf', 'logpdf'})
+
+        ndim = len(self._shape)
+        x_name, y_name = x, y
+        t_name, tlim = t[0], np.asarray(t[1:])
+        tlim = tlim[:, np.newaxis] if ndim else tlim
+
+        # pdf/logpdf are not valid for `t` because we can't easily invert them
+        message = (f'Argument `t` of `{self.__class__.__name__}.plot` "'
+                   f'must be one of {valid_t}')
+        if y_name not in valid_xy:
+            raise ValueError(message)
+
+        message = (f'Argument `x` of `{self.__class__.__name__}.plot` "'
+                   f'must be one of {valid_xy}')
+        if x_name not in valid_xy:
+            raise ValueError(message)
+
+        message = (f'Argument `y` of `{self.__class__.__name__}.plot` "'
+                   f'must be one of {valid_xy}')
+        if y_name not in valid_xy:
+            raise ValueError(message)
+
+        # This could just be a warning
+        message = (f'`{self.__class__.__name__}.plot` was called on a random '
+                   'variable with at least one invalid shape parameters. When '
+                   'a parameter is invalid, no plot can be shown.')
+        if self._any_invalid:
+            raise ValueError(message)
+
+        # We could automatically ravel, but do we want to? For now, raise.
+        message = ("To use `plot`, distribution parameters must be "
+                   "scalars or arrays with one or fewer dimensions.")
+        if ndim > 1:
+            raise ValueError(message)
+
+        try:
+            import matplotlib.pyplot as plt  # noqa: F401, E402
+        except ModuleNotFoundError as exc:
+            message = ("`matplotlib` must be installed to use "
+                       f"`{self.__class__.__name__}.plot`.")
+            raise ModuleNotFoundError(message) from exc
+
+        ax = plt.gca() if ax is None else ax
```

Comment:
> I think we already have quite some similar helpers. The custom message this would bring is not that necessary IMO.

Which?

> Also consider moving after qlim to have everything grouped.

I consider this to be grouped with the try-except. I don't see how it would be grouped with `qlim`.

Reply:
Ah right, but then could be in the try part. But it's ok like this, just a pref.

For the plot we have e.g. in spatial `_held_figure`, in stats in `FitResult`, `EmpiricalDistributionFunction.plot`. Maybe other places.

Reply:
If you factor out a function, I'll use it.